File: 3_claude_sol1.py
def main():
    M = int(input())
    m = M
    primes = []
    p = 2
    while p * p <= m:
        if m % p == 0:
            primes.append(p)
            while m % p == 0:
                m //= p
        p += 1
    if m > 1:
        primes.append(m)

    count = 0
    for x in range(1, M):
        for y in range(1, M):
            ok = True
            for p in primes:
                if x % p != 0 and y % p != 0:
                    ok = False
                    break
            if ok:
                count += 1
    print(count)

main()

File: 3_claude_sol1.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int M;
    cin >> M;
    vector<int> primes;
    int m = M;
    for(int p = 2; (long long)p * p <= m; p++){
        if(m % p == 0){
            primes.push_back(p);
            while(m % p == 0) m /= p;
        }
    }
    if(m > 1) primes.push_back(m);

    long long count = 0;
    for(int x = 1; x <= M - 1; x++){
        for(int y = 1; y <= M - 1; y++){
            bool ok = true;
            for(int p : primes){
                if(x % p != 0 && y % p != 0){
                    ok = false;
                    break;
                }
            }
            if(ok) count++;
        }
    }
    cout << count << endl;
    return 0;
}
